gt/lua_lpm.c

```diff
@@ -0,0 +1,395 @@
+		sizeof(struct in6_addr));
+	*(struct in6_addr *)cdata = ip_addr.ip.v6;
+
+	lua_pushinteger(l, ret);
+
+	return 2;
+}
+
+static int
+l_new_lpm(lua_State *l)
+{
+	struct rte_lpm_config lpm_conf;
+	struct rte_lpm **lpm;
+	static rte_atomic32_t identifier = RTE_ATOMIC32_INIT(0);
+
+	memset(&lpm_conf, 0, sizeof(lpm_conf));
+
+	/* First argument must be a Lua number. */
+	lpm_conf.max_rules = luaL_checknumber(l, 1);
+
+	/* Second argument must be a Lua number. */
+	lpm_conf.number_tbl8s = luaL_checknumber(l, 2);
+
+	if (lua_gettop(l) != 2)
+		luaL_error(l, "Expected two arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	lpm = (struct rte_lpm **)lua_newuserdata(l, sizeof(struct rte_lpm *));
+	*lpm = init_ipv4_lpm("gt_", &lpm_conf, rte_socket_id(),
+		rte_lcore_id(), rte_atomic32_add_return(&identifier, 1));
+
+	luaL_getmetatable(l, "gt_lpm");
+	lua_setmetatable(l, -2);
+
+	return 1;
+}
+
+static int
+l_lpm_add(lua_State *l)
+{
+	int ret;
+
+	/* First argument must be of type struct rte_lpm **. */
+	struct rte_lpm **lpm =
+		(struct rte_lpm **)luaL_checkudata(l, 1, "gt_lpm");
+
+	/*
+	 * Second argument must be a Lua number.
+	 * @ip must be in network order.
+	 */
+	uint32_t ip = luaL_checknumber(l, 2);
+
+	/* Third argument must be a Lua number. */
+	uint8_t depth = luaL_checknumber(l, 3);
+
+	/* Fourth argument must be a Lua number. */
+	uint32_t label = luaL_checknumber(l, 4);
+
+	if (lua_gettop(l) != 4)
+		luaL_error(l, "Expected four arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	ret = rte_lpm_add(*lpm, ip, depth, label);
+	if (ret < 0) {
+		luaL_error(l, "lpm: failed to add network policy [ip: %u, depth: %%hhu, label: %u] to the lpm table at %s!",
+			ip, depth, label, __func__);
+	}
+
+	lua_pushinteger(l, 0);
+
+	return 1;
+}
+
+static int
+l_lpm_del(lua_State *l)
+{
+	int ret;
+
+	/* First argument must be of type struct rte_lpm **. */
+	struct rte_lpm **lpm =
+		(struct rte_lpm **)luaL_checkudata(l, 1, "gt_lpm");
+
+	/*
+	 * Second argument must be a Lua number.
+	 * @ip must be in network order.
+	 * */
+	uint32_t ip = luaL_checknumber(l, 2);
+
+	/* Third argument must be a Lua number. */
+	uint8_t depth = luaL_checknumber(l, 3);
+
+	if (lua_gettop(l) != 3)
+		luaL_error(l, "Expected three arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	ret = rte_lpm_delete(*lpm, ip, depth);
+
+	lua_pushinteger(l, ret);
+
+	return 1;
+}
+
+static int
+l_lpm_lookup(lua_State *l)
+{
+	int label;
+
+	/* First argument must be of type struct rte_lpm **. */
+	struct rte_lpm **lpm =
+		(struct rte_lpm **)luaL_checkudata(l, 1, "gt_lpm");
+
+	/*
+	 * Second argument must be a Lua number.
+	 * @ip must be in network order.
+	 */
+	uint32_t ip = luaL_checknumber(l, 2);
+
+	if (lua_gettop(l) != 2)
+		luaL_error(l, "Expected two arguments, however it got %d arguments",
+			lua_gettop(l));
+
+	label = lpm_lookup_ipv4(*lpm, ip);
+
+	lua_pushinteger(l, label);
+
+	return 1;
```

Comment:
Drop variable `label`, combine both lines above, and drop the blank line before `return 1;`.